### ISSUE TITLE
stratified train_test_split

### ISSUE THREAD

**Issue by lacava** [lacava/feat#202]
add option to train_test_split that stratifies the splits so that the training and test sets have the same proportion of each class as the original data. 

one way to do this:

- split the original data up by class label into sets
- run train test split on each of the sets
- append the training sets and append the test sets to get X_t and X_v

**Comment by lacava** [comment]
this should address issue #174

**Comment by tilakhere** [comment]
commit https://github.com/lacava/feat/commit/a758929fd849bf84826f310a198427f507aba370

instead of splitting data into different sets the split is done using keeping track of sets of indices of each of the different classes.

Issue #174 still exists...trying to resolve it
